### ISSUE TITLE
Latest insiders does not recognise active pyenv virtualenv

### ISSUE THREAD

**Issue by albireox** [microsoft/pylance-release#4806]
The latest insiders seems to pick the base Python version when in a pyenv virtualenv (not sure if this is more generalised), resulting in imports that cannot be resolved

```console
2023-09-07 02:31:45.453 [info] [Info  - 2:31:45 AM] (24817) Pylance language server 2023.9.11 (pyright d58286a4) starting
2023-09-07 02:31:45.453 [info] [Info  - 2:31:45 AM] (24817) Server root directory: /home/gallegoj/.vscode-server-insiders/extensions/ms-python.vscode-pylance-2023.9.11/dist
2023-09-07 02:31:45.458 [info] [Info  - 2:31:45 AM] (24817) Starting service instance "main"
2023-09-07 02:31:45.465 [info] [Info  - 2:31:45 AM] (24817) Auto format for strings enabled
2023-09-07 02:31:45.495 [info] [Info  - 2:31:45 AM] (24817) Setting pythonPath for service "main": "/home/sdss5/opt/pyenv/versions/3.10.9/bin/python3.10"
2023-09-07 02:31:45.496 [info] [Info  - 2:31:45 AM] (24817) Setting pythonPath for service "main": "/home/sdss5/opt/pyenv/versions/3.10.9/bin/python3.10"
2023-09-07 02:31:45.498 [info] [Info  - 2:31:45 AM] (24817) Loading pyproject.toml file at /home/sdss5/software/lvmguider/main/pyproject.toml
2023-09-07 02:31:45.502 [info] [Info  - 2:31:45 AM] (24817) Pyproject file "/home/sdss5/software/lvmguider/main/pyproject.toml" has no "[tool.pyright]" section.
2023-09-07 02:31:45.528 [info] [Info  - 2:31:45 AM] (24817) Assuming Python version 3.10
2023-09-07 02:31:45.597 [info] [Info  - 2:31:45 AM] (24817) Found 33 source files
```

For comparison the same project with the stable version

```console
2023-09-07 02:30:26.415 [info] [Info  - 2:30:26 AM] (23990) Pylance language server 2023.9.10 (pyright c70adefc) starting
2023-09-07 02:30:26.416 [info] [Info  - 2:30:26 AM] (23990) Server root directory: /home/gallegoj/.vscode-server-insiders/extensions/ms-python.vscode-pylance-2023.9.10/dist
2023-09-07 02:30:26.425 [info] [Info  - 2:30:26 AM] (23990) Starting service instance "main"
2023-09-07 02:30:26.430 [info] [Info  - 2:30:26 AM] (23990) Auto format for strings enabled
2023-09-07 02:30:26.474 [info] [Info  - 2:30:26 AM] (23990) Setting pythonPath for service "main": "/home/sdss5/opt/pyenv/versions/3.10.9/envs/lvmguider-main/bin/python"
2023-09-07 02:30:26.474 [info] [Info  - 2:30:26 AM] (23990) Setting pythonPath for service "main": "/home/sdss5/opt/pyenv/versions/3.10.9/envs/lvmguider-main/bin/python"
2023-09-07 02:30:26.477 [info] [Info  - 2:30:26 AM] (23990) Loading pyproject.toml file at /home/sdss5/software/lvmguider/main/pyproject.toml
2023-09-07 02:30:26.480 [info] [Info  - 2:30:26 AM] (23990) Pyproject file "/home/sdss5/software/lvmguider/main/pyproject.toml" has no "[tool.pyright]" section.
2023-09-07 02:30:26.511 [info] [Info  - 2:30:26 AM] (23990) Assuming Python version 3.10
2023-09-07 02:30:26.599 [info] [Info  - 2:30:26 AM] (23990) Found 33 source files
2023-09-07 02:30:32.446 [info] [Info  - 2:30:32 AM] (23990) [IDX(2)] Long operation: index execution environment /home/sdss5/software/lvmguider/main (4764ms)
2023-09-07 02:30:32.595 [info] [Info  - 2:30:32 AM] (23990) [IDX(2)] Long operation: index packages /home/sdss5/software/lvmguider/main (4834ms)
```

This seems to happen with either the stable or insiders version of the Python extension, which does appear to identify the virtual environment correctly.

**Comment by rchiodo** [comment]
Thanks for the issue. This should depend entirely on the Python extension. We don't pick anything. They tell us what the python path is. 

Can you do two things?

Turn on trace level output logging for pylance.

```
"python.analysis.logLevel": "Trace"
```

And upload the log again.

Then also upload the log for the Python extension.

**Comment by albireox** [comment]
Here's the language server trace with both the latest pylance and Python extension

[pylance.log](https://github.com/microsoft/pylance-release/files/12551331/pylance.log)

The Python extension seems to reference the right environment (`/home/sdss5/opt/pyenv/versions/3.10.9/envs/lvmguider-main/bin/python`) and I get the right version in the bottom right section in the status bar. Everything else in the Python extension seems correct, but of course if the environment is not selected correctly that will be mostly obvious in pylance.

**Comment by rchiodo** [comment]
This error here I believe is related to the Python extension updating their in memory representation of the interpreter:

```
2023-09-07 16:04:58.941 [warning] Retry failed, storage update failed for key PYTHON_GLOBAL_STORAGE_KEYS
```

Which might be this issue here: https://github.com/microsoft/vscode-python/issues/21635

What's the version of the Python extension?

**Comment by albireox** [comment]
Here are the logs for pylance and the Python extensions with the log level set to trace.

[python_extension.log](https://github.com/microsoft/pylance-release/files/12551719/python_extension.log)
[pylance.log](https://github.com/microsoft/pylance-release/files/12551722/pylance.log)

I see these messages

```console
2023-09-07 16:58:12.964 [debug] Cached data exists KeyPrefix=Cache_Method_Output_b._selectIfLaunchedViaActivatedEnv-Args=true
```

but not sure if something else is amiss. The version of the Python extension is `v2023.17.12501006`

**Comment by rchiodo** [comment]
It looks like the python extension log wasn't uploaded correctly. It's 0KB.

**Comment by albireox** [comment]
My bad, forgot to CMD+S :)

[python_extension.log](https://github.com/microsoft/pylance-release/files/12552166/python_extension.log)

**Comment by luabud** [comment]
I am also able to reproduce this but using the WSL extension

My Python and Pylance logs: 
[pythonwsllogs.txt](https://github.com/microsoft/pylance-release/files/12552446/pythonwsllogs.txt)
[pylancewsllogs.txt](https://github.com/microsoft/pylance-release/files/12552447/pylancewsllogs.txt)

I have a .venv selected but it's resolving to /user/bin/python3.11 instead

**Comment by rchiodo** [comment]
Okay hopefully I can repro. The logs look like Python is returning the expected environment, so not sure how we're getting the wrong one.

**Comment by myuanz** [comment]
Reproduced on 2023.9.10 and 2023.9.11, the expected Python is a poetry interpreter derived from pyenv

[Pylance.txt](https://github.com/microsoft/pylance-release/files/12555317/Pylance.txt)
[Python.txt](https://github.com/microsoft/pylance-release/files/12555318/Python.txt)

**Comment by chrisk314** [comment]
I'm also facing this problem. Also using WSL2 in Windows 11 (Ubuntu). I noticed today that all my python environments are broken in all my VS Code Workspaces. When I open any python script I now have unresolved import errors for dependencies which are definitely present in the activated Python env. Interestingly I can setup launch configurations and these do run with the correct interpreter and import the dependencies correctly.

**Comment by alejoar** [comment]
This is happening to me as well, VS Code 1.82.

**Comment by rchiodo** [comment]
I can repro with WSL. Should mean we can fix it soon.

**Comment by rchiodo** [comment]
This issue has been fixed in prerelease version 2023.9.12, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#2023912-8-September-2023-PreRelease)
